code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -740,7 +740,7 @@
 	var/list/obj/machinery/camera/visible = list()
 	for (var/datum/camerachunk/CC in eyeobj.visibleCameraChunks)
 		for (var/obj/machinery/camera/C in CC.cameras)
-			if (!C.can_use() || get_dist(C, eyeobj) > 7)
+			if (!C.can_use() || get_dist(C, eyeobj) > 7 || iscyborg(C.loc))

```

Comment:
There should be a var in the camera for if that camera can light up instead of a type check on its location.